File: gitlab-nic/code-samples-main/API/Python/list-domains-paginated.py
#!/usr/bin/env python3

# Retrieves the list of all your domains

import requests
import json

import afnic

PAGE_SIZE=100 # The maximum value, over 100, you'll get a 400 HTTP error.

s = requests.session()
s.headers.update(afnic.get_headers())
got_all = False
page = 0
total_domains = None
domains = 0
while not got_all:
    response = s.get("https://api-sandbox.nic.fr/v1/domains?pageSize=%i&page=%i" % \
                            (PAGE_SIZE, page))
    body = json.loads(response.text)
    if response.status_code != 200:
        print(body)
        raise Exception("HTTP status code for list is %s" % response.status_code)
    total_domains = body["totalElements"]
    for domain in body["content"]:
        print(domain["name"])
        domains += 1
    print("%i domains retrieved" % domains)
    if domains >= total_domains or len(body["content"]) == 0:
        got_all = True
    else:
        page += 1
s.close()

File: gitlab-nic/code-samples-main/API/Go/list-domains-paginated.go
/* Retrieves the list of your domains. */

package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

const page_size = 5 /* The maximum value. Over 100, you'll get a 400
   /* HTTP error. */

func main() {
	got_all := false
	domains := 0
	page := 0
	for !got_all {
		url := fmt.Sprintf("https://api-sandbox.nic.fr/v1/domains?pageSize=%d&page=%d", page_size, page)
		request, err := http.NewRequest(http.MethodGet,
			url,
			nil)
		if err != nil {
			panic(err)
		}
		for name, value := range headers() {
			request.Header.Add(name, value)
		}
		response2, err := http.DefaultClient.Do(request)
		if err != nil {
			panic(err)
		}
		if response2.StatusCode != 200 {
			panic(fmt.Sprintf("Wrong HTTP response for list of domains: %s", response2.Status))
		}
		body, err := ioutil.ReadAll(response2.Body)
		if err != nil {
			panic(err)
		}
		type Domains struct {
			Name string /* We ignore the other members */
		}
		type Result struct {
			Content       []Domains
			TotalElements int `json:"totalElements"`
			/* We ignore the other members */
		}
		var result Result
		err = json.Unmarshal(body, &result)
		if err != nil {
			panic(err)
		}
		if len(result.Content) == 0 {
			break
		}
		for _, domain := range result.Content {
			domains += 1
			fmt.Printf("%s\n", domain.Name)
		}
		if domains >= result.TotalElements {
			got_all = true
			break
		} else {
			page += 1
		}
	}
	fmt.Printf("%d domains retrieved\n", domains)
}
